logs_ingest/logging.py

```diff
@@ -15,28 +15,54 @@
 import logging
 import os
 
+LOG_THROTTLING_LIMIT_PER_CALLER = 10
+log_call_count = dict()
+
 script_directory = os.path.dirname(os.path.realpath(__file__))
 version_file_path = os.path.join(script_directory, "version.txt")
 with open(version_file_path) as version_file:
     _version = version_file.readline()
     _version_tag = f"[{_version}] "
 
 
-def exception(msg, *args, **kwargs):
+def exception(msg, caller: str, *args, **kwargs):
+    if check_if_caller_exceeded_limit(caller):
+        return
     logging.exception(_version_tag + msg, *args, **kwargs)
 
 
-def error(msg, *args, **kwargs):
+def error(msg, caller: str, *args, **kwargs):
+    if check_if_caller_exceeded_limit(caller):
+        return
     logging.error(_version_tag + msg, *args, **kwargs)
 
 
-def warning(msg, *args, **kwargs):
+def warning(msg, caller: str, *args, **kwargs):
+    if check_if_caller_exceeded_limit(caller):
+        return
     logging.warning(_version_tag + msg, *args, **kwargs)
 
 
 def info(msg, *args, **kwargs):
     logging.info(_version_tag + msg, *args, **kwargs)
 
 
-def debug(msg, *args, **kwargs):
-    logging.debug(_version_tag + msg, *args, **kwargs)
\ No newline at end of file
+def debug(msg, caller: str, *args, **kwargs):
+    if check_if_caller_exceeded_limit(caller):
+        return
+    logging.debug(_version_tag + msg, *args, **kwargs)
```

Comment:
If we're not throttling info logs, I think we shouldn't throttle debug logs either. As far as I understand, debug and info logs are more "predictable" (they're not used in exceptional situations) and so they don't require throttling?